src/input.rs

```diff
@@ -569,6 +566,24 @@ mod tests {
         assert_eq!(err.to_string().contains(filename), true);
     }
 
+    #[test]
+    fn test_input_gzip() {
+        use std::io::Cursor;
+        let data = Cursor::new(
+            // echo '' | gzip -cf | od -A n -t u1 -w40| sed -e 's/^  *//; s/  */, /g'
+            vec![31, 139, 8, 0, 0, 0, 0, 0, 0, 3, 227, 2, 0, 147, 6, 215, 50, 1, 0, 0, 0],
+        );
+        let mut input = Input::open_stream(&PathBuf::from("test.log.gz"), Box::new(data)).unwrap();
+        let mut buf = [0; 32];
+        let result = input.stream.read(&mut buf);
+        assert!(result.is_ok());
+        let got = result.unwrap();
+        if got != 1 {
```

Comment:
`if got != 1` checks are not needed in rust, it is more convenient to use `assert_eq!` macro.
It prints both expressions' values in case of check failure.

Reply:
Thanks.
I know zero rust, it would be way easier in Go :)